packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import path from 'path'
+import glob from 'glob'
+
+/**
+ * @class OverridesResolverPlugin
+ *
+ *  This plugin provides the "Overrides" behavior of the Template Extensibility feature,
+ *  allowing third party implementations that depend on an npm module for the base implementation
+ *  and then overriding only specific files
+ */
+class OverridesResolverPlugin {
+    /**
+     *
+     * @param options
+     * @param {string} options.overridesDir path to application base
+     * @param {string[]} options.overrides paths to overrides
+     * @param {string} options.projectDir path to project directory
+     */
+    constructor(options) {
+        this.overridesDir = options.overridesDir || ''
+        this.extends = options.extends || []
+        this.projectDir = options.projectDir
+        this._allSearchDirs = [this.projectDir + this.overridesDir, ...this.extends]
```

Comment:
We always search `overridesDir` for a given import (e.g. `retail-react-app`) first, so we add it here